src/heatmapInternal.ts

```diff
@@ -0,0 +1,150 @@
+// Internal class that encapsulates private properties and methods
```

Comment:
@nswamy14 I was facing an issue when I was using Heatmap class directly. Since most of the properties and methods were public, they were breaking the api contract of the module, hence created this internal class to satisfy the public api contract without exposing all the internal logic.